packages/core/src/external/analytics/posthog.ts

```diff
@@ -18,7 +19,53 @@ export interface EventMessageV1 extends IdentifyMessageV1 {
 
 const defaultPostHogApiKey = Config.getPostHogApiKey();
 const groupType = "customer";
+let posthogClient: PostHog | undefined;
 
+export function initPostHog(apiKey?: string): PostHog | undefined {
+  const postHogApiKey = apiKey ?? defaultPostHogApiKey;
+  if (!postHogApiKey) return undefined;
+
+  return new PostHog(postHogApiKey);
+}
+
+function getPosthogClientInstance(): PostHog | undefined {
+  return posthogClient ?? initPostHog(defaultPostHogApiKey);
+}
+
+export function captureAnalytics(params: EventMessageV1): void {
+  const posthogClient = getPosthogClientInstance();
+  if (!posthogClient) return;
+
+  const enrichedParams = {
+    ...params,
+    properties: {
+      ...(params.properties ?? {}),
+      environment: Config.getEnvType(),
+      platform: "oss-api",
+    },
+    groups: { [groupType]: params.distinctId },
+  };
+
+  posthogClient.capture(enrichedParams);
+}
+
+/**
+ * Can send multiple analytics events in one bulk request when it calls to shutdown.
+ * @link - https://posthog.com/docs/libraries/node#using-in-a-short-lived-process-like-aws-lambda
+ */
+export async function captureAnalyticsAsync(
+  events: EventMessageV1 | EventMessageV1[]
+): Promise<void> {
+  if (!posthogClient) return;
+  toArray(events).forEach(captureAnalytics);
+
+  // Needed to send requests to PostHog in lambda
+  await posthogClient.shutdown();
+}
```

Comment:
I'd keep only the `captureAnalytics` and have a `closePosthog()` function that should be called at the end of the lambda's execution, in a `finally` section.

This keeps the code benefitting from their batching, while making sure the analytics are send almost always (unless the lambda times out - which, BTW, we could deal with by using `Promise.race`).

Fine if we want to lave it to another PR, maybe even ticket it and add a comment w/ the ticket number.

I'd just try to avoid creating new functions, the shared instance seems like an expected pattern from Posthog's docs and thus should be safe.